### ISSUE TITLE
Unexpected behavior for `evolve` when `transformations` contains a null value.

### ISSUE THREAD

**Issue by tansongyang** [ramda/ramda#1832]
The following code produces unexpected output ([repl](http://goo.gl/PgJSY7)):

```JavaScript
var o  = {a: 123, b: 456, c: 789};
var transformations = {
  a: null,
  b: undefined,
  c: 'foo',
};
R.evolve(transformations, o); // -> {"a": {}, "b": 456, "c": 789}
```

The value of `a` became an empty object instead of its original value, `123`.

Looking at the [code](https://github.com/ramda/ramda/blob/master/src/evolve.js#L38), it seems like this could be resolved by changing the check from `type === 'object'` to `type && type === 'object'` (unless this is expected behavior).

**Comment by buzzdecafe** [comment]
yeah that sucks. your solution seems fine to me though. would you like to do a PR?

**Comment by CrossEye** [comment]
Yes, I think that should be fixed.  I think you will have to do `transformation && type === 'object'` instead, though.  Javascript, in its infinite, er, wisdom, has decided that `typeof(null); //=> 'object'`.

**Comment by tansongyang** [comment]
@CrossEye Good catch! @buzzdecafe I'd be glad to do PR. Will let you know when it's done.

**Comment by tansongyang** [comment]
@buzzdecafe Let me know if there are any issues. I took a best guess at what the style should be.
